modular_skyrat/master_files/code/modules/clothing/towels.dm

```diff
@@ -0,0 +1,523 @@
+
+	if(!cleaning_themselves)
+		to_chat(user, span_notice("You finish drying [target_mob] up with \the [src]."))
+
+	var/water_to_remove = min(max(-target_mob.fire_stacks, 0), free_space)
+
+	if(!water_to_remove)
+		return
+
+	reagents.add_reagent(/datum/reagent/water, water_to_remove)
+	target_mob.set_wet_stacks(0, remove_fire_stacks = FALSE)
+
+	set_wet(TRUE, update_visuals = shape != TOWEL_FOLDED)
+
+	if(shape == TOWEL_FOLDED)
+		change_towel_shape(user, TOWEL_USED, TRUE)
+
+
+/obj/item/towel/attack_self(mob/user, modifiers)
+	. = ..()
+
+	/// Initializing this only once to avoid having to do it every time
+	var/static/list/datum/radial_menu_choice/worn_options = list()
+
+	if(!length(worn_options))
+		for(var/variant in list(TOWEL_FULL, TOWEL_WAIST, TOWEL_HEAD))
+			var/datum/radial_menu_choice/option = new
+			var/image/variant_image = image(icon = TOWEL_OBJ_ICON, icon_state = "[base_icon_state]-[variant]")
+
+			option.image = variant_image
+			worn_options[capitalize(variant)] = option
+
+	var/choice = show_radial_menu(user, src, worn_options, require_near = TRUE, tooltips = TRUE)
+
+	if(!choice)
+		return
+
+	change_towel_shape(user, lowertext(choice))
+
+
+/obj/item/towel/attackby(obj/item/attacking_item, mob/user, params)
+	. = ..()
+
+	if(!(attacking_item.tool_behaviour == TOOL_WIRECUTTER || attacking_item.get_sharpness()))
+		return
+
+	if(flags_1 & HOLOGRAM_1) // Just in case there's ever holographic towels.
+		return
+
+	var/obj/item/stack/sheet/cloth/shreds = new (get_turf(src), TOWEL_CLOTH_AMOUNT)
+
+	if(!QDELETED(shreds)) //stacks merged
+		transfer_fingerprints_to(shreds)
+		shreds.add_fingerprint(user)
+
+	to_chat(user, span_notice("You tear [src] up into cloth."))
+	qdel(src)
+
+
+/obj/item/towel/pre_attack_secondary(atom/target, mob/living/user, params)
+	. = ..()
+
+	if(!istype(target, /obj/item/reagent_containers/cup/bucket))
+		return
+
+	if(!reagents.total_volume)
+		to_chat(user, span_warning("\The [src] is dry, you can't squeeze anything out!"))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+	var/obj/item/reagent_containers/cup/bucket/target_bucket = target
+
+	if(target_bucket.reagents.total_volume >= target_bucket.reagents.maximum_volume)
+		to_chat(user, span_warning("[target] is full!"))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+	transfer_towel_reagents_to(target_bucket, reagents.total_volume, user, loss_factor = SQUEEZING_DISPERSAL_RATIO, make_used = TRUE) // If it didn't have enough space, oh well, you lost like 3/4th of what was in the towel anyway, there's just even more loss that way. Doesn't really matter.
+
+	to_chat(user, span_notice("You wring the liquid out of [src], transferring some of it to [target]."))
+
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+
+/obj/item/towel/AltClick(mob/user)
+	. = ..()
+
+	if(. == FALSE)
```

Comment:
why not just !. instead

Reply:
Because it can return `null` too, which isn't what I want to catch in this case. I explicitly want to catch only `FALSE`.